### DISCUSSION TITLE
Optional seed parameter for RNG-based features

### DISCUSSION THREAD

**Discussion by jrego43** [LazaroHurtado/micrograd-rs#6] [Ideas]
I think this would make it easier for developers to predictably test things like weight initialization and it would increase the project's overall flexibility. Thoughts?

**Comment by LazaroHurtado** [comment]
I agree that increasing test coverage is should be a top priority and encouraged. When I first created this module I was hesitant about how it should be done so I moved towards more flashy features, which is bad on my end.

After some thought, allowing users to pass a defined seed might become useful for some developers and increase flexibility like you said so I support this idea. Might also be useful to create a separate function from the existing `sample` to at least abstract this idea behind one method, and incorporate a new `sample_seeded` method which would probably require moving the match statement to a helper function to not write duplicate code.
